### ISSUE TITLE
AutoDMG v1.10b587 fails to create file macOS Catalina 10.15 19A583 

### ISSUE THREAD

**Issue by alby696** [MagerValp/AutoDMG#241]
## Description of issue

After a while the build fails with:
Installation of 'InstallInfo.plist' failed with return code 1. See the log for details.

I put the catalina installer both on the desktop and in a different folder than in a nas, but the problem remains the same.

## Expected behavior

AutoDMG should build a DMG consisting of macOS Catalina

## Actual behavior

see above

## Steps to reproduce

1. Start AutoDMG v1.10b587 on macOS Catalina 10.15 19A583
2. Drag the macOS Catalina 10.15 19A583 Installer App onto the AutoDMG-window
3. Click „Build“
4. Wait a while

## Log file

[AutoDMG 2019-10-08 22.17.log](https://github.com/MagerValp/AutoDMG/files/3704358/AutoDMG.2019-10-08.22.17.log)

**Comment by erikng** [comment]
You need to put the app in a DMG and then drag it onto auto DMG for it to work.

Thanks,
Erik Gomez
________________________________

**Comment by lp122158** [comment]
[AutoDMG 2019-10-08 18.21.log](https://github.com/MagerValp/AutoDMG/files/3704758/AutoDMG.2019-10-08.18.21.log)

**Comment by lp122158** [comment]
I also notice that the autodmg shows unknown build 19A583 which was downloaded directly from the app store. The actual installer works when ran locally.

**Comment by alby696** [comment]
I created a disk image in both extended macOs and APFS and copied into the installer, but when I put it on autodmg it tells me that it fails to activate it and gives me error.

**Comment by lp122158** [comment]
Question, I am only seeing this issue executing this on a laptop however, it works on a iMac. Not sure if its a build failed error. Photo below shows same build number but different Computer Model 
![IMG_1280](https://user-images.githubusercontent.com/56324534/66523811-d03f3580-eabe-11e9-8509-09f270d3cbf2.JPG)

**Comment by alby696** [comment]
I'm using an iMac 21.5" late 2015, now I test it with a laptop. Do I need to create an extended apfs or macos disk image and then copy it in and try to feed it to autodmg?

**Comment by lp122158** [comment]
Just take the .app installer to autodmg and see if it builds the image for you using autodmg. I had a friend using a similar iMac to try to create the .dmg using autodmg and it worked fine.

**Comment by jelockwood** [comment]
@MagerValp 
If I put the Catalina installer in a dmg then it seems to run successfully. It does then when deployed via DeployStudio result in a bootable Mac. (With Catalina.)

I would say that both the documentation needs to be updated to reflect the need for the installer to be in a dmg and for AutoDMG to immediately flag this problem if a user drags a 'naked' installer on to the app.

PS. Whilst the installer (from inside a dmg) initially shows the correct Catalina icon after AutoDMG has analysed it, it reverts to a generic app icon.

**Comment by danielooi** [comment]
Hi guys, the trick of putting the installer .app file inside the .dmg and then drag it from the dmg container  to the AutoDMG works.

**Comment by HowardGyton** [comment]
Hi,

I just found this thread.  I too had the same issue with the naked .app files.  I followed advice I found to create the DMG bu using the Disk Utility, creating a DMG from an folder, and pointing at a folder that has the installer app. inside.

However, when I then drag that DMG onto AutoDMG I get the error:

Invalid source
Couldn't find system version.

I accepted the defaults for DMG creation:

Encryption: None
Image Format: Compressed

**Comment by jelockwood** [comment]
@HowardGyton 
Firstly with regards to an earlier comment, AutoDMG is now showing the icon for Catalina.

Secondly I use the following script to download the Catalina installer and this script automatically places it in a disk image and this resulting disk image works with AutoDMG.

https://github.com/munki/macadmin-scripts/blob/master/installinstallmacos.py

I would however suggest that it would be worthwhile an additional error check be added to AutoDMG so if it detects the installer is not being dragged from a disk image it immediately warns this rather than allowing you to run the attempt and only much later failing.

**Comment by HowardGyton** [comment]
I managed to get my disk image created and copied across to the desktop of an external disk with a bootable copy of 10.15.2.  I booted a Mac from this disk, destroyed the parititions on the internal disk and then "attempted" to use "Disk Utility" to copy the image across.....lots of error 22s.

And creating a volume that was in APFS format simply generated a message stating that to copy an image to an APFS formatted disk, the source image had to be APFS, and as far as I know it is!

In the end, after some more reading, most people seem to have given up on DU and use CCC instead.  This worked, and I now have a deployable image, with Centrify embeeded for AD intergration, all from an external disk.

Has anyone had any luck getting "Disk Utility" to deploy an APFS AutoDMG image?

**Comment by Lotusshaney** [comment]
Don’t user Disk Utility to deploy. Instead use ASR

**Comment by jelockwood** [comment]
@HowardGyton 
I have now managed to do the following. Use AutoDMG to create a disk image with various packages run and included. I have then managed to use DeployStudio Runtime running from a Catalina boot drive to restore that image. I get the impression that this approach does not do any firmware updates.

The trick I discovered was to ensure the 'Macintosh HD Data' volume is mounted before doing the DeployStudio Runtime restore image workflow as most of the restored stuff ends up in that rather than 'Macintosh HD'.

You also have to give things like DeployStudio Runtime 'Full Disk' permission so it can write to the entire 'Macintosh HD'.

However I have now moved on and I am able to do the following.

1. Boot from a USB that contains the Catalina installer as created using the createinstallmedia command as per https://support.apple.com/en-gb/HT201372 I use this to format the Mac and then install a fresh Catalina OS
2. Then reboot from the Catalina boot drive I have DeployStudio Runtime on
3. Use DeployStudio Runtime to login to the DeployStudio repo
4. Run a workflow which configures the target, runs installer packages, scripts, etc.

This then results in a fully configured bootable Mac running Catalina.

Note: There is a bug in DeployStudio 1.7.8 which results in /Library having the incorrect permissions, I therefore have another workflow which solely runs a script which resets the permissions and then immediately reboots.
